ImGuiFileDialog.cpp

```diff
@@ -1,72 +1,100 @@
-#include "ImGuiFileDialog.h"
+// From https://github.com/aiekick/ImGuiFileDialog
+//
+// Modified by AlastairGrowcott@yahoo.com to work on Posix systems.
+
+
+#include <sys/types.h>
+#include <dirent.h>
```

Comment:
On Linux the header is simply "dirent.h". The Windows versions of these Posix functions are very different and so are the types. On Linux we need to include sys/types.h too. I can't remember why I needed limits.h and stdlib.h.

algorithm.h has simply moved up so that similar types of header are grouped together.